### DISCUSSION TITLE
Render a mesh directly

### DISCUSSION THREAD

**Discussion by LiamGallagher737** [bevyengine/bevy#5162] [Q&A] [answered]
Is there a way to render/draw a mesh for a single frame without need to spawn an entity? The use case for for a visual gizmo system. 

I've attempted trying to use `wgpu` and `winit` directly but was unable to get that working and I feel that isn't how I should be going about it.

Any help would be greatly appreciated.

**Comment by komadori** [comment, marked as answer]
I would be cautious of whatever intuition you have regarding the cost of creating Entities and so on. However, yes, you can:

You can add a system to the `RenderStage::Queue` stage of the `RenderApp` and obtain a `RenderPhase<Opaque3d>` or similar into which you can `add()` a request to draw your mesh for a given frame. You will still need a way to communicate what should be drawn to your custom rendering system. This is done via systems running in the `RenderStage::Extract` stage, which normally convert normal-world Entities into RenderApp Entities, but you could use your own data structure in a Resource to do this rather than Entities.

Bevy's animate_shader* example is the best(/only?) example of queuing your own rendering in the RenderApp.

* https://github.com/bevyengine/bevy/blob/main/examples/shader/animate_shader.rs
